File: CPP/ByRound/2169/2169_D1_Removal_of_a_Sequence_Easy_Version.cpp
// Problem: CF 2169 D1 - Removal of a Sequence (Easy Version)
// https://codeforces.com/contest/2169/problem/D1

/*
Code Purpose:
This program solves the problem of determining the k-th element in a sequence after performing x removal operations.
Each operation removes elements at positions y, 2*y, 3*y, ..., m*y where m*y <= current sequence length.
The algorithm simulates the reverse process to find the k-th element efficiently without constructing the full sequence.

Algorithms/Techniques:
- Simulation with reverse thinking: For each removal step, we compute how the position of an element changes.
- Mathematical manipulation to avoid simulating all removals step-by-step.

Time Complexity: O(x), where x is the number of operations; each operation involves constant time math.
Space Complexity: O(1), only using a few variables for computation.

*/
#include <iostream>

using namespace std;
#define int long long
signed main() {
  int t;
  cin >> t;
  while (t--) {
    int x, y, z;
    cin >> x >> y >> z;
    if (y == 1) {
      cout << -1 << endl;
      continue;
    }
    int curr = z;  // Start from the k-th position and trace back
    while (x--) {
      // Compute the new position after one removal operation
      // curr += y * ((curr - 1) / (y - 1)) - (y - 1) * ((curr - 1) / (y - 1));
      // Simplify above line to:
      curr += ((curr - 1) / (y - 1));  // This line effectively updates position
    }
    cout << ((curr > 1e12) ? -1 : curr) << endl;
  }
}


// https://github.com/VaHiX/CodeForces/

File: Python/ByTier/D/2169_D1_Removal_of_a_Sequence_Easy_Version.py
# Problem: CF 2169 D1 - Removal of a Sequence (Easy Version)
# https://codeforces.com/contest/2169/problem/D1

"""
Purpose: 
  This code solves the problem of determining the k-th element in a sequence 
  after repeatedly removing elements at positions y, 2*y, 3*y, ..., m*y 
  for x iterations. The algorithm uses mathematical reasoning to avoid 
  simulating the full process, which is too slow due to the large size of the 
  initial sequence (up to 10^12).

Algorithms/Techniques:
  - Mathematical simulation: Instead of simulating removals, we compute
    how many elements remain after each step using the formula:
    new_length = current_length - current_length // y
  - Reverse calculation of position: We simulate the inverse process to 
    determine where the k-th element ends up in the final sequence.

Time Complexity: O(x), since we iterate x times for each test case.
Space Complexity: O(1), as only a constant amount of extra space is used.
"""

s = set()
for _ in range(int(input())):  # Read number of test cases
    x, y, k = map(int, input().split())  # Read x, y, k for current test case
    curn = 10**12  # Start with full sequence of length 10^12
    
    # Simulate the removal process x times
    for i in range(x):
        curn -= curn // y  # Remove every y-th element, update current length
    
    # If k is beyond the final sequence length, output -1
    if curn < k or y == 1:
        print(-1)
        continue

    ans = k  # Start with the position in the final sequence
    
    # Reverse simulate to compute what index in the original sequence maps to k-th position
    for i in range(x):
        ans += (ans - 1) // (y - 1)
    
    print(ans)  # Output the result


# https://github.com/VaHiX/CodeForces/